resources/views/platform/components/beaten-games-leaderboard/meta-panel/index.blade.php

```diff
@@ -0,0 +1,38 @@
+@props([
+    'allSystems' => null,
+    'gameKindFilterOptions' => [],
+    'selectedConsoleId' => null,
+    'selectedAllowHacks' => true,
+])
+
+<script>
+function handleConsoleChanged(event) {
+    const newQueryParamValue = event.target.value;
+    window.updateUrlParameter('filter[system]', newQueryParamValue);
```

Comment:
Changing filters does not reset the current page. This can result in misleading "There aren't any rows matching your current filter criteria." errors, even though there would be rows on the first page.

To reproduce: go to page 2 or 3, then select a system with very few awards (like TV Games Computer)

Reply:
Addressed in latest, these JS functions responsible for managing the filter query params will now also simultaneously reset the page number to 1.

https://github.com/RetroAchievements/RAWeb/pull/1862/commits/b113ee264709ee466b4c1087f32adcabd84b3d15#diff-a5605035ffaeff131236bb669ee4856fc0a03efb14891106d3d6ab8f482af09cR10-R20

Reply:
Now I get an `Attempt to read property "User" on null` error following the steps I listed above. I seem to get it switching to any system, even on the first page.

Reply:
Resolved in https://github.com/RetroAchievements/RAWeb/pull/1862/commits/38a30ffa76b61f7f2e0da092e868e649ad0ea141.